.github/workflows/actions-ci.yml

```diff
@@ -433,64 +243,109 @@ jobs:
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - uses: actions/setup-go@v4
         with:
           go-version: '>=1.18'
-      - name: Setup CMake
-        uses: threeal/cmake-action@v1.3.0
-        with:
-          generator: Ninja
-          c-compiler: clang-${{ matrix.clangversion }}
-          cxx-compiler: clang++-${{ matrix.clangversion }}
-          options: FIPS=${{ matrix.fips }} CMAKE_BUILD_TYPE=Release
+      - name: Setup Clang
+        if: startsWith(matrix.compiler, 'clang')
+        run: |
+          echo "CC=clang" >> $GITHUB_ENV
+          echo "CXX=clang++" >> $GITHUB_ENV
+      - name: Setup GCC
+        if: startsWith(matrix.compiler, 'gcc')
+        run: |
+          echo "CC=gcc" >> $GITHUB_ENV
+          echo "CXX=g++" >> $GITHUB_ENV
+      - name: Setup Build
+        if: matrix.fips == 0
+        run:
+          cmake -G Ninja -B ./build -DCMAKE_BUILD_TYPE=Release
+      - name: Setup FIPS Build
+        if: matrix.fips == 1
+        run:
+          cmake -G Ninja -B ./build -DCMAKE_BUILD_TYPE=Release -DFIPS=1
       - name: Build Project
         run: cmake --build ./build --target all
       - name: Run tests
         run: cmake --build ./build --target run_tests
 
-  clang-19-sanity:
-    if: github.repository_owner == 'aws'
-    needs: [ sanity-test-run ]
+  pedantic-tests:
+    env:
+      GOFLAGS: "-buildvcs=false"
     strategy:
       fail-fast: false
       matrix:
-        fips:
-          - "0"
-          - "1"
-    runs-on: ubuntu-22.04
+        fips: [0, 1]
+        compiler:
+          - 'gcc13'
+          - 'gcc15'
+          - 'clang19'
+          - 'clang20'
+        exclude:
+          - fips: 1
+            compiler: 'gcc15'
+          - fips: 1
+            compiler: 'clang20'
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/mattkretz/cplusplus-ci/${{ matrix.compiler }}
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - uses: actions/setup-go@v4
         with:
           go-version: '>=1.18'
-      - name: Install clang-19
+      - name: Setup Clang
+        if: startsWith(matrix.compiler, 'clang')
         run: |
-          wget https://apt.llvm.org/llvm.sh
-          chmod u+x llvm.sh
-          sudo ./llvm.sh 19
-      - name: Setup CMake
-        uses: threeal/cmake-action@v1.3.0
-        with:
-          generator: Ninja
-          c-compiler: clang-19
-          cxx-compiler: clang++-19
-          options: FIPS=${{ matrix.fips }} CMAKE_BUILD_TYPE=Release
+          echo "CC=clang" >> $GITHUB_ENV
+          echo "CXX=clang++" >> $GITHUB_ENV
+      - name: Setup GCC
+        if: startsWith(matrix.compiler, 'gcc')
+        run: |
+          echo "CC=gcc" >> $GITHUB_ENV
+          echo "CXX=g++" >> $GITHUB_ENV
+      - name: Setup Build
+        if: matrix.fips == 0
+        run:
+          cmake -G Ninja -B ./build -DCMAKE_BUILD_TYPE=Release CMAKE_C_FLAGS=-pedantic CMAKE_CXX_FLAGS=-pedantic
```

Comment:
Can this be turned on by default in our CMakeLists.txt? Why is this a special CI dimension?